_LowLevelCode/cpp/utility/environment.cpp

```diff
@@ -0,0 +1,42 @@
+#include "utility/environment.h"
+
+
+namespace Horace {
+namespace Utility {
+namespace Environment {
+/*
+ * Retrieve an environment variable from the system.
+ * This function is necessary as `std::getenv` raises security warnings on
+ * Windows. Also note, GCC does not implement `getenv_s`.
+ *
+ * @param key :: The key of the environment variable
```

Comment:
missing a note on the return parameter saying it returns empty if variable not found

Reply:
would it make it "easier" for users if the API was `string get_variable(string key, string default)` where default (not empty) was returned if the key doesn't exist.

That way anyone calling can be sure the value is populated correctly on return and doesn't need to add a `if(isempty(value)) { value = default }` block of code  Thinking specifically of needing `.` for HORACE_ROOT.